### DISCUSSION TITLE
Why does attribute access result in a new SELECT?

### DISCUSSION THREAD

**Discussion by csirmazbendeguz** [sqlalchemy/sqlalchemy#11119] [Usage Questions] [answered]
I have noticed when I access an ORM object's attribute, a new SELECT query is made.

```
with session.begin():
    user = User(email="alasdai.haynes@example.com", password="abc")
    session.add(user)

print("user.email", user.email)  # SELECT users.id AS ...
```

I am confused why this is happening, and can I disable it?

**Comment by zzzeek** [comment, marked as answer]
this is called lazy loading.  background on this is available at https://docs.sqlalchemy.org/en/20/glossary.html#term-lazy-loading 

the reason your code example above would lazy load is because you have [expire_on_commit](https://docs.sqlalchemy.org/en/20/orm/session_basics.html#session-committing) left at its default value of `True`.

**Comment by csirmazbendeguz** [comment]
Thanks, that makes sense, didn't know about the expire on commit!
